File: flipt-client-java/src/main/java/io/flipt/client/models/Response.java
package io.flipt.client.models;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.util.Optional;
import lombok.Builder;
import lombok.Value;

@Value
@Builder
@JsonDeserialize(builder = Response.ResponseBuilder.class)
public class Response {
  @JsonProperty("type")
  String type;

  @JsonProperty("variant_evaluation_response")
  Optional<VariantEvaluationResponse> variantEvaluationResponse;

  @JsonProperty("boolean_evaluation_response")
  Optional<BooleanEvaluationResponse> booleanEvaluationResponse;

  @JsonProperty("error_evaluation_response")
  Optional<ErrorEvaluationResponse> errorEvaluationResponse;

  @lombok.experimental.SuperBuilder
  public static class ResponseBuilder {}
}

File: flipt-client-kotlin-android/src/main/java/io/flipt/client/models/Response.kt
package io.flipt.client.models

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class Response(
    @SerialName("type")
    val type: String,

    @SerialName("variant_evaluation_response")
    val variantEvaluationResponse: VariantEvaluationResponse?,

    @SerialName("boolean_evaluation_response")
    val booleanEvaluationResponse: BooleanEvaluationResponse?,

    @SerialName("error_evaluation_response")
    val errorEvaluationResponse: ErrorEvaluationResponse?
)